Force destroy buckets

saliceti opened the pull request:
This PR allows deleting the content of S3 and GCE buckets on a terraform destroy action.
This is really useful when using test environments. It may be dangerous for production, but if one uses _destroy_ on a production environment we would probably have bigger problems anyway.

**How to review**
_terraform destroy_ on GCE and AWS should delete everything, including storage.

**Note**
If the environment was created _before_ this update, you may have to apply it again before using destroy.

saliceti committed:
force_destroy parameter for GCE and AWS buckets

Avoid the following error:
Error deleting S3 Bucket: BucketNotEmpty: The bucket you tried to delete is not empty
	status code: 409, request id: []

Allows deleting the content of S3 and GCE buckets on a terraform destroy action.
This is really useful when using test environments. It is potentially dangerous as the bucket resource may be implicitly destroyed and recreated during a simple apply.
It can be specified via the force_destroy variable, which is false by default.

dcarley commented:
> This PR allows deleting the content of S3 and GCE buckets on a terraform destroy action.
> This is really useful when using test environments. It may be dangerous for production, but if one uses destroy on a production environment we would probably have bigger problems anyway.

This is really good detail and should be included in body of each commit message, so that somebody looking at the repo history in the future can determine why we did this and whether we should later stop doing this. Can you rebase both commits to add it?

dcarley commented:
It might also be useful to describe in the commit why it's useful for test environments at the moment; that it results in an error when you run `terraform destroy` to remove an environment that you no longer want. You and I probably know that, but someone new to our team probably won't.

saliceti commented:
force_destroy is now optional, thanks @dcarley .

keymon committed:
Update README to explain apply limiting target

To change the force_deploy setting in the buckets you need to run apply again
limiting the target to the buckets. It is better to set the variable
during the initial provisioning.

This commit explains how set this variable in the initial provision and
how to change the setting if it was not set previously.

keymon commented:
For me works fine, but I updated the README to explain and suggest use this new option while provisioning test environment and how to change it if needed.

As I did a new commit, I don't qualify to merge... 

:+1: from me, though.

mtekel commented:
Works fine. Also tested updating just one resource.

dcarley commented:
> For me works fine, but I updated the README to explain and suggest use this new option while provisioning test environment and how to change it if needed.
> 
> As I did a new commit, I don't qualify to merge...

:+1:
